server/src/main/java/io/spine/server/tuple/package-info.java

```diff
@@ -19,42 +19,57 @@
  *
+ * <h1>Overview</h1>
+ * 
  * <p>Although tuples are <a href="https://github.com/google/guava/wiki/IdeaGraveyard#tuples-for-n--2">
  * considered harmful</a> in general, they are useful for describing types of several messages
  * returned by a method. Consider the following example.
  *
  * <p>The return value of the below method does not say much about the number and types
- * of returned event messages.
- * <pre>{@code
- *     {@literal @}Assign
- *     List<Message> on(CreateTask cmd) { ... }
- * }</pre>
- *
- * The below declaration gives both number and types of the events:
- * <pre>{@code
- *     {@literal @}Assign
- *     Pair<TaskCreated, TaskAssigned> on(CreateTask cmd) { ... }
- * }</pre>
- *
- * <p>It should re-iterated that the purpose of this package is limited to the scenarios
- * described above. Programmers are strongly discouraged from applying tuples for other purposes.
- *
- * <h1>Generic Types</h1>
- *
- * <p>Classes provided by this package can support up to 5 generic parameters. They are named from
- * {@code <A>} through {@code <E>}.
- *
- * <p>The first generic parameter {@code <A>} must always be a specific
- * {@link com.google.protobuf.Message Message} class.
- *
- * <p>Types from {@code <B>} through {@code <E>} can be either {@code Message} or
- * {@link java.util.Optional Optional}. See sections below for details.
- *
- * <h1>Basic Tuples</h1>
+ * of returned event messages. We just know that there may be more than one of them.
+ * <pre>
+ *    {@literal @}Assign
+ *     List&lt;EventMessage&gt; on(CreateTask cmd) { ... }
+ * </pre>
+ *
+ * The below declaration gives both the number and types of the returned events:
+ * <pre>
+ *    {@literal @}Assign
+ *     Pair&lt;TaskCreated, Optional&lt;TaskAssigned&gt;&gt; on(CreateTask cmd) { ... }
+ * </pre>
+ *
+ * <p>Why do we suggest returning a tuple (a {@code Pair} in the example above) instead of creating
+ * a message type which would aggregate those of interest as our colleagues from the Guava team
+ * <a href="https://github.com/google/guava/wiki/IdeaGraveyard#tuples-for-n--2">suggest</a>?
+ *
+ * <p>Messages returned from a handling method are loosely coupled. They may not
+ * be created together all the time. The example above tells that {@code TaskAssigned}
+ * may not happen upon the creation of the task. It's possible that a task can be assigned
+ * sometime later. In this case combining returned messages does not reflect the domain language.
+ * Returning a tuple tells a part of the story in terms of the domain language right in the method
+ * signature. Details of this story are obtained from the method body.
+ *
+ * <p>It should be re-iterated that the purpose of this package is limited to the scenarios
+ * of handling messages. Programmers are strongly discouraged from using this package for
+ * other purposes.
+ *
+ * <h1>Two groups of classes</h1>
+ * This package provides two groups of classes:
```

Comment:
I would have an empty line above. So that the Javadoc in its raw form reads better in the editor.